parser.d.ts

```diff
@@ -106,6 +107,8 @@ export type NonCapturingGroup<F extends Features = {}> = Base<"group"> & {
     | "negativeLookahead"
     | "negativeLookbehind";
   body: RootNode<F>[];
+  enablingFlags?: string;
```

Comment:
If these are only there when `modifiers` is true please can you use `_If` to make it only `undefined` when it’s not enabled? Similar to the bit below

Reply:
actually from a typescript point of view IMO this would be nicer if it was a nested object like
```
export type ModifierFlags = {
  enablingFlags: string,
  disablingFlags: string
}
```

and then

```suggestion
  enablingFlags?: ModifierFlags;
```

as that means there's just 2 states. It's either all there, or none of it there

Reply:
Oh and also if this is only possible for `behaviour: 'ignore'` would be good to structure so that this is only there for `ignore` too. So something like

```ts
export type NonCapturingGroup<F extends Features = {}> = Base<"group"> &
  (
    | {
        behavior:
          | "lookahead"
          | "lookbehind"
          | "negativeLookahead"
          | "negativeLookbehind";
        body: RootNode<F>[];
      }
    | ({
        behavior: "ignore";
        body: RootNode<F>[];
      } & _If<
        F["namedGroups"],
        {
          modifierFlags?: ModifierFlags;
        },
        {
          modifierFlags: undefined;
        }
      >)
  );

```

Reply:
amazing! thanks!